control-plane/controllers/configentry_controller.go

```diff
@@ -267,6 +268,13 @@ func (r *ConfigEntryController) ReconcileEntry(ctx context.Context, crdCtrl Cont
 		return r.syncSuccessful(ctx, crdCtrl, configEntry)
 	}
 
+	// For resolvers and splitters, we need to set the ClusterIP of the matching service to Consul so that transparent
+	// proxy works correctly. Do not fail the reconcile if assigning the virtual IP returns an error.
+	err = assignServiceVirtualIP(ctx, logger, consulClient, crdCtrl, req.NamespacedName, configEntry)
```

Comment:
Can we check if the config entry is a splitter or a resolver before we assign the virtual IP? It makes it easier to understand why we invoke this method in that case.

Reply:
We could but having the check inside of assignServiceVirtualIP makes the tests a little bit nicer imho. Otherwise the only test I can do is 'does the service have a cluster IP'.

Reply:
I think that is ok? from a readability perspective, it makes sense to "assign service virtual IP" only when a particular config entry needs that assignment vs moving the config entry filtering inside the method. you could potentially create a method "needsVirtualIPAssignment" that returns true for certain config entries and false for other and test it's behavior therby covering test scenarios?

Reply:
I like this idea!

Reply:
@thisisnotashwin At your suggestion, I added a 'needs' method which does clean it up a little bit.